### PR TITLE
Add comp alg and block size to sextant pack

### LINKED ISSUE #250: Add sextant parameters for mksquashfs's compression algorithm and block size
The supported compression algorithms by squashfs depend on the kernel configuration. Different targets might support different compression algorithms. For this reason, it makes sense to add a parameter to sextant to specify the compression algorithm used to pack the container rootfs. 

An other important parameter is the size of the compressed blocks of data. It has a influence both in the access speed as well as the size of the resulting file.

### PR DESCRIPTION
**Pull request by alfonsoros88**
closes #250

### CODE REVIEW

File: examples/build_examples.sh [hunk 1]
```diff
@@ -17,6 +17,13 @@ else
   PLATFORM=${1}
 fi
 
+if [ -z ${2+x} ]
+then
+  COMPRESSION_ALGORITHM="gzip"
+else
+  COMPRESSION_ALGORITHM=${2}
+fi
+
```
Comment (flxo) [inline examples/build_examples.sh:26]: I't probably time to document the two optional arguments for this script in the README.md

The enum can have a `Display` impl to be used when invoking `mksquashfs`

File: npk/src/npk.rs [hunk 2]
```diff
@@ -84,10 +84,32 @@ pub enum Error {
     },
 }
 
+/// Squashfs Options
+pub struct SquashfsOpts<'a> {
+    /// The compression algorithm used (default gzip)
+    pub comp: &'a str,
```
Comment (flxo) [inline npk/src/npk.rs:90]: Would be great to introduce a enum for the compression algorithm. This allows `clap` to hint if the argument is unknown.

File: sextant/src/main.rs [hunk 3]
```diff
@@ -34,6 +34,12 @@ enum Opt {
         /// Output directory
         #[structopt(short, long)]
         out: PathBuf,
+        /// Compression algorithm to use in squashfs (default gzip)
+        #[structopt(short, long)]
+        comp: Option<String>,
```
Comment (flxo) [inline sextant/src/main.rs:39]: Use a enum here instead of `String` and declare a `default_value`

**Review by flxo — CHANGES_REQUESTED** [review]
Think we can improve the CLI experience by typing the compression algorithm.

File: npk/src/npk.rs [hunk 4]
```diff
@@ -350,10 +359,64 @@ impl Builder {
...
+    pub comp: CompAlg,
+    /// Size of the blocks of data compressed separately
+    pub block_size: u32,
+}
+
+impl Default for SquashfsOpts {
+    fn default() -> Self {
+        SquashfsOpts {
+            comp: CompAlg::Gzip,
+            block_size: 131072,
```
Comment (flxo) [inline npk/src/npk.rs:406]: Where does this number come from?

File: npk/src/npk.rs [hunk 5]
```diff
@@ -350,10 +359,64 @@ impl Builder {
...
+        }
+    }
+}
+
+/// Squashfs Options
+pub struct SquashfsOpts {
+    /// The compression algorithm used (default gzip)
+    pub comp: CompAlg,
+    /// Size of the blocks of data compressed separately
+    pub block_size: u32,
```
Comment (flxo) [inline npk/src/npk.rs:399]: Could this be optional? If it's None take the default of `mksquashfs`

File: npk/src/npk.rs [hunk 6]
```diff
@@ -350,10 +359,64 @@ impl Builder {
     }
 }
 
+#[derive(Debug)]
+pub enum CompAlg {
```
Comment (flxo) [inline npk/src/npk.rs:363]: Rename this to `CompressionAlgorithm`

File: npk/src/npk.rs (hunk 4 above)
Comment (alfonsoros88) [reply npk/src/npk.rs:406]: That's the default value used by `mksquashfs`

**Pull request merged**